src/cmd-process.ts

```diff
@@ -147,22 +157,15 @@ export class CmdProcess {
 
     if (this.cp.stdout)
       this.cp.stdout.pipe(split()).on('data', (line: string) => {
-        if (this.opts.collectLogs) stdOutBuffer.push(line)
+        if (this.opts.collectLogs && !this._activeOutput) this._outputBuffer.push({ type: 'stdout', line })
         else console.log(this.autoAugmentLine(line))
         if (this.doneCriteria && this.doneCriteria.test(line)) this._finished.resolve()
       })
     if (this.cp.stderr)
       this.cp.stderr.pipe(split()).on('data', (line: string) => {
-        if (this.opts.collectLogs) stdErrBuffer.push(line)
+        if (this.opts.collectLogs && !this._activeOutput) this._outputBuffer.push({ type: 'stderr', line })
         else console.error(this.autoAugmentLine(line))
         if (this.doneCriteria && this.doneCriteria.test(line)) this._finished.resolve()
       })
-    if (this.opts.collectLogs)
```

Comment:
Why not place the code that logs the lines in an interleaved way here? This piece runs once per command process, on the process close

Reply:
Only one process can log to the console at the time.

When one process terminates, it cannot write all the output to the console, because another process might be writing to it already. So the output from process B would show up in the middle of the output from process A.

Process B needs to wait until its turn to write to the output. This is signalled by the call to `activateOutput`.

Reply:
Lets say A executes in 10 seconds and B executes in 5 seconds.

A will then start to write directly to the console. B will buffer its output not to mix it with the output from A.
When B is done, it still has to wait for A to complete before it can write all its output to the console.